cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -17,53 +15,229 @@
+                System.out.println("SBOM: " + json);
+            } break;
 
-        System.out.println("SBOM: " + json);
-    }
+            case "addMetadataProperty": {                     //This is MetaData--> Component --> Property -> name-value:
+                bom = readJSONfile();
+                bom = addMetadataProperty(bom, name, value);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
+
+            case "addComponent": {                              //This adds Component with component name
+                bom = readJSONfile();
+                bom = addComponent(bom, name);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
 
-    static Bom createTestBom() {
-        Bom bom = new Bom();
+            case "addComponentProp": {                             //This adds Components with name-value pairs to List
+                bom = readJSONfile();
+                bom = addComponentProperty(bom, name, value);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
 
-        Component comp1 = new Component();
-        comp1.setName("TestComponent");
-        comp1.setVersion("1.0.0");
-        comp1.setType(Component.Type.APPLICATION);
-        comp1.setAuthor("Adoptium");
+            case "addExtRef": {                                    //This adds external Reference
+                bom = readJSONfile();
+                bom = addExternalReference(bom, url, comment);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
 
-        bom.addComponent(comp1);
+            case "addComponentExtRef": {                                 //This adds external Reference to Component
+                bom = readJSONfile();
+                bom = addComponentExternalReference(bom, url, hashes, comment);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
+            default: {
+                System.out.println("Please enter a command.");
+            }
+        }
+    }
 
+    static Bom createBom(final String name, final String version) {        //Create SBOM, test.JSON file
+        bom = new Bom();
+        System.out.println(bom.getBomFormat());
+        comp.setName(name);
+        comp.setVersion(version);
+        comp.setType(Component.Type.APPLICATION);
+        comp.setGroup("Eclipse Temurin");
+        comp.setAuthor("Vendor: Adoptium");
+        bom.addComponent(comp);
```

Comment:
Not sure what this comp is being added here?

Reply:
Added this comp to show below data on top. 
    {
      "author" : "Vendor: Adoptium",
      "group" : "Eclipse Temurin",
      "name" : "Eclipse Temurin",
      "version" : "jdk11",
      "type" : "application"
    },
Now once we make multiple components, will move this data to top-level component.